dapp/contracts/loans/InvestorLedger.sol

```diff
@@ -41,9 +41,15 @@ library InvestorLedger {
             account.loanToken.allowance(trustee, this),
             account.totalLoanNeeded - account.totalAmountGathered
         );
+        require(
```

Comment:
This needs to go add the end of the function (after all state modification), otherwise you're risking recursive calling bugs.

Reply:
http://solidity.readthedocs.io/en/develop/security-considerations.html#re-entrancy